mpc/mpc-core/src/garble/label/mod.rs

```diff
@@ -1,54 +1,56 @@
+//! Types associated with wire labels
 
 mod digest;
+pub(crate) mod encoded;
 mod encoder;
 pub(crate) mod input;
 pub(crate) mod output;
 mod set;
-mod state;
 
-use mpc_circuits::{Circuit, GroupId, GroupValue, Input, Output, Value, WireGroup};
-use rand::{CryptoRng, Rng};
 use std::{
-    ops::{BitXor, Deref},
+    ops::{BitAnd, BitXor, Deref},
     sync::Arc,
 };
 
-use crate::{block::Block, garble::LabelError};
+use mpc_circuits::{Input, Output, Value};
+use rand::{CryptoRng, Rng};
+
+use crate::{block::Block, garble::EncodingError};
 
 pub use digest::LabelsDigest;
+pub use encoded::{Encoded, GroupDecodingInfo};
 pub use encoder::ChaChaEncoder;
 pub use output::OutputLabelsCommitment;
-pub use set::LabelsSet;
+pub use set::EncodedSet;
 
 /// A collection of full labels not associated with a wire group.
-pub type FullLabels = state::Full;
+pub type FullLabels = Labels<Full>;
 /// A collection of active labels not associated with a wire group.
-pub type ActiveLabels = state::Active;
+pub type ActiveLabels = Labels<Active>;
 
 /// Full input labels, ie contains both the low and high labels.
-pub type FullInputLabels = Labels<Input, state::Full>;
-/// Active input labels of a garbled circuit. These are the labels which the evaluator uses
+pub type FullEncodedInput = Encoded<Input, Full>;
+/// Active encoded input of a garbled circuit. These are the labels which the evaluator uses
 /// to evaluate the circuit.
-pub type ActiveInputLabels = Labels<Input, state::Active>;
-/// Input labels decoding information
-pub type InputLabelsDecodingInfo = LabelsDecodingInfo<Input>;
-/// Full output labels of a garbled circuit
-pub type FullOutputLabels = Labels<Output, state::Full>;
-/// Active output labels of a garbled circuit. These are the labels which the evaluator derives
+pub type ActiveEncodedInput = Encoded<Input, Active>;
+/// Input decoding information.
+pub type InputDecodingInfo = GroupDecodingInfo<Input>;
+/// Full encoded output of a garbled circuit.
+pub type FullEncodedOutput = Encoded<Output, Full>;
+/// Active encoded output of a garbled circuit. These are the labels which the evaluator derives
 /// as a result of the circuit evaluation.
-pub type ActiveOutputLabels = Labels<Output, state::Active>;
-/// Output labels decoding information
-pub type OutputLabelsDecodingInfo = LabelsDecodingInfo<Output>;
-
-/// A complete set of full input labels
-pub type FullInputLabelsSet = LabelsSet<Input, state::Full>;
-/// A complete set of full output labels
-pub type FullOutputLabelsSet = LabelsSet<Output, state::Full>;
-/// A complete set of active input labels
-pub type ActiveInputLabelsSet = LabelsSet<Input, state::Active>;
-/// A complete set of active output labels
-pub type ActiveOutputLabelsSet = LabelsSet<Output, state::Active>;
+pub type ActiveEncodedOutput = Encoded<Output, Active>;
+/// Output decoding information.
+pub type OutputDecodingInfo = GroupDecodingInfo<Output>;
+
+/// A complete set of full encoded inputs
```

Comment:
"full encoded inputs and active encoded inputs" is more obscure vs "full input labels and active input labels".